R/object_name_linters.R

```diff
@@ -129,7 +129,8 @@ make_object_linter <- function(fun) {
     keep_indices <- which(
       !is_operator(names) &
         !is_known_generic(names) &
-        !is_base_function(names)
+        !is_base_function(names) &
+        !is_special_function(names)
```

Comment:
Is there a better name than `special`? maybe `special_hook`? "special" seems a bit generic. If no better option springs to mind, just go ahead.

Reply:
(also for the NEWS)

Reply:
I thought if we ever need to add additional functions to this list for whatever reason, we might not want to blow up the conditional and instead just update the list of special functions with more comments on why they're special.

Reply:
OK that makes sense. I was going to suggest improving the NEWS but I think that's done now too.